File: composeApp/src/commonMain/kotlin/net/matixmedia/og_meals/domain/model/Meal.kt
package net.matixmedia.og_meals.domain.model

import kotlinx.datetime.LocalDate


data class Meal(
    val id: Long,
    val title: String,
    val alternativeTitle: String?,
    val price: Double,
    val studentPrice: Double?,
    val guestPrice: Double?,
    val date: LocalDate,
    val sustainabilityCo2: Int?,
    val canteenId: Long,
    val nutritionalInfo: NutritionalInfo?,
    val allergens: List<MealAllergen>,
    val additives: List<MealAdditive>,
    val features: List<MealFeature>,
    val category: MealCategory?,
    val mealImages: List<MealImage>
)

data class NutritionalInfo(
    val kj: Int,
    val kcal: Int,
    val fat: Double,
    val saturatedFat: Double,
    val carbohydrates: Double,
    val sugar: Double,
    val protein: Double,
    val salt: Double
)

data class MealAllergen(
    val id: Long,
    val name: String?,
    val shortName: String?
)

data class MealAdditive(
    val id: Long,
    val name: String?,
    val shortName: String?
)

data class MealFeature(
    val id: Long,
    val name: String?,
    val shortName: String?,
    val showInOverview: Boolean?,
    val showInFilter: Boolean?
)

data class MealCategory(
    val id: Long,
    val name: String?
)

data class MealImage(
    val aiSuggested: Boolean,
    val url: String
)

File: app/types/meal.ts
import type { MealReviewStats } from './review'

export interface MealImageDto {
  url: string
  aiSuggested: boolean
  suggestedBasedOnMealId?: number | null
}

export interface MealFeature {
  id: number
  name?: string | null
  shortName?: string | null
  orderInApp?: number | null
  rgbColor?: string | null
  showInFilter?: boolean | null
  showInOverview?: boolean | null
}

export interface MealAllergen {
  id: number
  name?: string | null
  shortName?: string | null
}

export interface MealAdditive {
  id: number
  name?: string | null
  shortName?: string | null
}

export interface MealNutritionalInfo {
  kj: number | null
  kcal: number | null
  fat: number | null
  saturatedFat: number | null
  carbohydrates: number | null
  sugar: number | null
  protein: number | null
  salt: number | null
}

export interface MealCategory {
  id: number
  name?: string | null
  unifiedName?: string | null
}

export interface Meal {
  id: number
  plu?: string | null
  title: string
  hash?: string | null
  alternativeTitle?: string | null
  price: number | null
  studentPrice: number | null
  guestPrice?: number | null
  date: string
  sustainabilityCo2?: number | null
  canteenId: number
  nutritionalInfo?: MealNutritionalInfo | null
  allergens?: MealAllergen[]
  additives?: MealAdditive[]
  images: MealImageDto[]
  features: MealFeature[]
  category?: MealCategory | null
  reviewStats: MealReviewStats
}
